lrlex/src/lib/parser.rs

```diff
@@ -221,7 +232,7 @@ impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
                 self.start_states.push(start_state);
             }
         }
-        Ok(off + line_len)
+        self.parse_ws(i)
```

Comment:
The purpose of this change is the `let src = "%S X ";` test in, which I had forgotten to re-add.
Otherwise this accepts `X` as a state name and skips right over the `En Space` at the end of the line without producing an error, there might be a better error we could do here, if test something like`line_end == self.parse_ws(i)`.

But I need a better understanding of what has gone awry in that test.

https://github.com/softdevteam/grmtools/pull/340/commits/7635044013a546b6c90e145c7cbcc650d3de337a

Reply:
Should be fixed in https://github.com/softdevteam/grmtools/pull/340/commits/54825ad126c18ee75def82d06d38986e79d6f8d6 This is an issue which was much more prevalent in the code than I had anticipated, essentially we need to use `trim_matches` instead of `trim` and friends, which work on the unicode whitespace.